sdks/java/core/src/main/java/com/google/cloud/dataflow/sdk/transforms/windowing/Window.java

```diff
@@ -599,6 +600,33 @@ public void validate(PCollection<T> input) {
       return output.setWindowingStrategyInternal(outputStrategy);
     }
 
+    @Override
+    public void populateDisplayData(DisplayData.Builder builder) {
+      builder
+          .add("windowFn", windowFn.getClass())
+          .include(windowFn);
+
+      if (trigger != null) {
+        builder.add("trigger", trigger.toString());
+      }
+
+      if (mode != null) {
+        builder.add("accumulationMode", mode.toString());
+      }
+
+      if (allowedLateness != null) {
+        builder.add("allowedLateness", allowedLateness);
+      }
+
+      if (closingBehavior != null) {
+        builder.add("closingBehavior", closingBehavior.toString());
+      }
+
+      if (outputTimeFn != null) {
```

Comment:
Most of these we need the explicit null check because we're dereferencing (`closingBehavior.toString()`)